TheaterSchedule.BLL/Services/AdminsPostService.cs

```diff
@@ -0,0 +1,61 @@
+﻿using AutoMapper;
+using System.Collections.Generic;
+using Entities.Models;
+using TheaterSchedule.BLL.DTOs;
+using TheaterSchedule.BLL.Interfaces;
+using TheaterSchedule.DAL.Interfaces;
+
+namespace TheaterSchedule.BLL.Services
+{
+    public class AdminsPostService : IAdminsPostService
+    {
+        private ITheaterScheduleUnitOfWork theaterScheduleUnitOfWork;
+        private IAdminsPostRepository adminsPostRepository;
+
+        private IMapper mapper;
+
+        public AdminsPostService(ITheaterScheduleUnitOfWork theaterScheduleUnitOfWork, IAdminsPostRepository adminsPostRepository)
+        {
+            this.adminsPostRepository = adminsPostRepository;
+            this.theaterScheduleUnitOfWork = theaterScheduleUnitOfWork;
+
+            var config = new MapperConfiguration(cfg =>
+            {
+                cfg.CreateMap<AdminsPost, AdminsPostDTO>()
+                   .ReverseMap();
+            });
+
+            mapper = config.CreateMapper();
+        }
+
+        public void AddNewPost(AdminsPostDTO post)
+        {
+            adminsPostRepository.Add(mapper.Map<AdminsPostDTO, AdminsPost>(post));
+            theaterScheduleUnitOfWork.Save();
+        }
+
+        public List<AdminsPostDTO> GetAllPersonalPosts(int id)
+        {
+            var posts = adminsPostRepository.GetAllPersonalById(id);
+
+            if (posts == null)
```

Comment:
looks like this if does nothing.

Reply:
There were a problem in some version of Automapper when it was throwing exception on null. I discovered it while working on TodoList.
Seems like it is not a problem anymore. 
So I agree this code is redundant